automatic-curtain/cpp/automatic-curtain.cpp

```diff
@@ -53,6 +53,36 @@ void checkRoot(void)
 	return;
 }
 
+int initPlatform(int& aPinRotary, int& aPinLight, int& dPinButton, int& i2cPort)
+{
+	//Update these values as per the hardware board being used.
+	aPinRotary = 1,
+	aPinLight = 2,
+	dPinButton = 4,
+	i2cPort = 0;
+	// check which board we are running on
+	Platform platform = getPlatformType();
+	switch (platform) {
+	case INTEL_UP2:
+#ifdef USING_GROVE_PI_SHIELD //512 offset needed for the shield
+		aPinRotary += 512;     // A1 Connector
+		aPinLight += 512;      // A2 Connector
+		dPinButton += 512;     // D4 Connector
+		break;
+#else
+		cerr << "Not using Grove provide your pinout here" << endl;
```

Comment:
why is this error case? Grove shield is a a hardware that may or may not be added to the up2. If they are not using grove shield then the pins don't need to be offset by 512 and the rest would work. 
So pin would be 1,2,4 and 0 and sample would work.

Reply:
Actually it wouldn't, since the UP2 uses a RPI header layout rather than an Arduino one and has no AIO.